src/main/java/org/fundacionjala/coding/BarCode.java

```diff
@@ -0,0 +1,35 @@
+package org.fundacionjala.coding;
+
+/**
+ * this class check de code bar.
+ * Created by Administrator on 8/29/2017.
+ */
+public class BarCode {
+
+    /**
+     * This function check is the code bar is true.
+     * @param codBar number de code bal.
+     * @return boolean.
+     */
+    public boolean checkCodeBar(String codBar) {
+        int checkSum = Integer.parseInt(codBar.substring(codBar.length() - 1));
+        int sum = 0;
+        for (int i = 1; i < codBar.length() - 1; i++) {
+            if (i % 2 == 0) {
+                sum += (int) (codBar.charAt(i));
+            } else {
+                sum += ((int) (codBar.charAt(i)) * 3);
+            }
+        }
+
+        if (sum % 10 == 0) {
+            return (checkSum == 0) ? true : false;
+
+        } else {
+            return ((10 - (sum % 10)) == checkSum) ? true : false;
+
+        }
```

Comment:
this if else can be refactored in two lines of code
in order to avoid return true or false.

Reply:
Try this code.
```java
        int sum = 0;
        for (int i = 1; i < codBar.length() - 1; i++) {
            int number = (codBar.charAt(i));
            sum += i % 2 == 0 ? number : number * 3;
        }
        int checksum = sum % 10 == 0 ? 0 : 10 - (sum % 10);
        int lastNumber = Integer.parseInt(codBar.substring(codBar.length() - 1));
        return checksum == lastNumber;
```